### PR TITLE
chore: limit CI runs

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Apart of #24754 

## Rules
- always run on PRs ready-for-review and protected branched
- allow CI on any branch run when explicitly opt'd into when commit message includes "run ci"
- allow CI when pipeline is manually triggered from CircleCI UI via `Trigger Pipeline` button

## Changes
- DRAFT PR 
	- [skipped by default](https://app.circleci.com/pipelines/github/cypress-io/cypress/46008/workflows/c6768b88-6544-4aaa-b32c-173bdad078dc/jobs/1934258)
	- **NOTE:** Draft PR to PR Ready for Review does not trigger a new CI build. To kick off a build after marking a draft as ready will require either manually kicking off in CircleCI UI via `Trigger Pipeline` or by push a new commit with any message.
- WIP branch 
  - [skipped by default](https://app.circleci.com/pipelines/github/cypress-io/cypress/46004/workflows/14e9925d-5db6-47ae-bf5b-75a75b3d2a7d/jobs/1934209)
  - [runs when commit message includes 'run ci'](https://app.circleci.com/pipelines/github/cypress-io/cypress/45999/workflows/9495dfd8-086c-438c-a2f3-0699f02ce082/jobs/1934119)

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -0,0 +1,2898 @@
+version: 2.1
```
Comment (emilyrohrbough) [inline .circleci/workflows.yml:1]: This file is 1:1 with the previous `config.yml` file. CircelCI uses `config.yml` by default. This change is to simple limit kicking off the workflows. Over time,  plan to make enhancements to the dynamic configuration to simplify this setup.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,2907 +1,70 @@
 version: 2.1
+setup: true
```
Comment (lmiller1990) [inline .circleci/config.yml:2]: Any tips to review this PR? There's a lot of red and green, but I suspect the code changes are quite minimal?

Comment (emilyrohrbough) [reply .circleci/config.yml:2]: workflows.yml should be 1:1 copy of config.yml in develop, so copy & paste and see nothing changed? I was bummed the diff on config.yml was so rough. config.yml is all new content

Comment (lmiller1990) [reply .circleci/config.yml:2]: Right, ok, so I just need to really look at https://github.com/cypress-io/cypress/blob/1ac9a10649f283841688fc54d665553052506c29/.circleci/config.yml#L26 for new code.

Bash is gnarly, if it works it works -- cool hack on downloading the workflows.yml: https://github.com/cypress-io/cypress/blob/1ac9a10649f283841688fc54d665553052506c29/.circleci/config.yml#L61

Comment (lmiller1990) [reply .circleci/config.yml:2]: I see little to no risk in this change, so :heavy_check_mark:

**Review by lmiller1990 — APPROVED** [review]

**Review by astone123 — APPROVED** [review]
Nice! We should probably announce a summary of these changes in Slack just so that people know what's going on 👍🏻

**Pull request merged**